tiledb/sm/rest/rest_client.h

```diff
@@ -66,7 +66,8 @@ class RestClient {
       stats::Stats* parent_stats,
       const Config* config,
       ThreadPool* compute_tp,
-      const std::shared_ptr<Logger>& logger);
+      const std::shared_ptr<Logger>& logger,
+      ContextResources& resources);
```

Comment:
It occurs to me that adding the ContextResources here probably removes the need for all the other arguments except maybe `config`. When you create the "Fix RestClient constructor" story, can you just make a note about this, please?